flink/v1.16/flink/src/main/java/org/apache/iceberg/flink/sink/BaseDeltaTaskWriter.java

```diff
@@ -94,7 +94,11 @@ public void write(RowData row) throws IOException {
         writer.delete(row);
         break;
       case DELETE:
-        writer.delete(row);
+        if (upsert) {
+          writer.deleteKey(keyProjection.wrap(row));
+        } else {
```

Comment:
nit: for consistency with other case clauses. maybe no `else`?
```
if (upsert) {
...
}
...
break;
````

Reply:
@stevenzwu I don't think we should be consistent with other cases, the update case has to delete the key and write the row so it doesn't have an else clause, but here we only need to delete once.

Reply:
that's true.